docs/book/src/images/worker-machines-resources.plantuml

```diff
@@ -0,0 +1,46 @@
+@startuml worker-machines-resources
```

Comment:
The suggestions below assume the general relationships (for any, or no bootstrap provider) rather than specifically the Kubeadm bootstrapper. 

I think this could be generalized even further by replacing references to KubeadmConfig and KubeadmConfigTemplate with generic naming, such as BarBootstrapTemplate/BarBootstrapConfig.

Reply:
I changed to use generic naming for boostrap provider CRDs, such as BarBoostrapConfig and BarBootstrapControlPlane at 95d7f93.